saleor/order/utils.py

```diff
@@ -212,3 +215,35 @@ def restock_fulfillment_lines(fulfillment):
         if line.order_line.variant and line.order_line.variant.track_inventory:
             increase_stock(
                 line.order_line.variant, line.quantity, allocate=True)
+
+
+def validate_shipping_method(order):
+    """Validates `order.shipping_method`.
+
+    Clears shipping attributes for a draft order if its `shipping_method`
+    is invalid after the draft order has been updated. A draft order is invalid
+    if either its price or weight is not within the range of the price or
+    weight of the `ShippingMethod` that the user has selected.
+    """
+    if order.is_shipping_required():
```

Comment:
I wonder, if there will be a case, when draft order is shipping required but has no shipping_method assigned yet. What do you think?

Reply:
I think it depends on where or not ```shipping_method``` is *required* to be selected by the user when they have selected a product that requires shipping -- I do not think there is any such constraint. It would not hurt to program defensively here and add that case. 

Should I cover the case?

Reply:
Tbh I'd cover it, better safe than sorry